samples/snippets/type_system_test.py

```diff
@@ -0,0 +1,357 @@
+# limitations under the License.
+
+import pandas.testing
+
+from bigframes import dtypes
+
+
+def test_type_system_examples() -> None:
+    # [START bigquery_dataframes_type_sytem_timestamp_local_type_conversion]
+    import pandas as pd
+
+    import bigframes.pandas as bpd
+
+    s = pd.Series([pd.Timestamp("20250101")])
+    assert s.dtype == "datetime64[ns]"
+    assert bpd.read_pandas(s).dtype == "timestamp[us][pyarrow]"
+    # [END bigquery_dataframes_type_sytem_timestamp_local_type_conversion]
+
+    # [START bigquery_dataframes_type_system_pyarrow_preference]
+    import datetime
+
+    import pandas as pd
+
+    import bigframes.pandas as bpd
+
+    s = pd.Series([datetime.date(2025, 1, 1)])
+    s + pd.Timedelta(hours=12)
+    # 0	2025-01-01
+    # dtype: object
+
+    bpd.read_pandas(s) + pd.Timedelta(hours=12)
+    # 0    2025-01-01 12:00:00
+    # dtype: timestamp[us][pyarrow]
+    # [END bigquery_dataframes_type_system_pyarrow_preference]
+    pandas.testing.assert_series_equal(
+        s + pd.Timedelta(hours=12), pd.Series([datetime.date(2025, 1, 1)])
+    )
+    pandas.testing.assert_series_equal(
+        (bpd.read_pandas(s) + pd.Timedelta(hours=12)).to_pandas(),
+        pd.Series([pd.Timestamp(2025, 1, 1, 12)], dtype=dtypes.DATETIME_DTYPE),
+        check_index_type=False,
+    )
+
+    # [START bigquery_dataframes_type_system_simple_json]
+    import db_dtypes
+    import pandas as pd
+
+    import bigframes.pandas as bpd
+
+    json_data = [
+        "1",
+        '"str"',
+        "false",
+        '["a",{"b":1},null]',
+        '{"a":{"b":[1,2,3],"c":true}}',
+        None,
+    ]
+    bpd.Series(json_data, dtype=pd.ArrowDtype(db_dtypes.JSONArrowType()))
+    # 0                               1
+    # 1                           "str"
+    # 2                           false
+    # 3              ["a",{"b":1},null]
+    # 4    {"a":{"b":[1,2,3],"c":true}}
+    # 5                            <NA>
+    # dtype: extension<dbjson<JSONArrowType>>[pyarrow]
+    # [END bigquery_dataframes_type_system_simple_json]
+    pandas.testing.assert_series_equal(
+        bpd.Series(
+            json_data, dtype=pd.ArrowDtype(db_dtypes.JSONArrowType())
+        ).to_pandas(),
+        pd.Series(json_data, dtype=pd.ArrowDtype(db_dtypes.JSONArrowType())),
+        check_index_type=False,
+    )
+
+    # [START bigquery_dataframes_type_system_mixed_json]
+    import db_dtypes
+    import pandas as pd
+    import pyarrow as pa
+
+    import bigframes.pandas as bpd
+
+    list_data = [
+        [{"key": "1"}],
+        [{"key": None}],
+        [{"key": '["1","3","5"]'}],
+        [{"key": '{"a":1,"b":["x","y"],"c":{"x":[],"z":false}}'}],
+    ]
+    pa_array = pa.array(list_data, type=pa.list_(pa.struct([("key", pa.string())])))
+    bpd.Series(
+        pd.arrays.ArrowExtensionArray(pa_array),
+        dtype=pd.ArrowDtype(
+            pa.list_(pa.struct([("key", db_dtypes.JSONArrowType())])),
```

Comment:
Same here. I think we'll always allow this for backwards compatibility but we might convert this to the arrow built-in json_ type in future.

CC @chelsea-lin 

Maybe we suggest folks use `bigframes.dtypes.JSON_ARROW_TYPE` instead?